crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -0,0 +1,929 @@
+                log_id,
+                segment_index,
+            } => {
+                if matches!(self.logs_state.get(&log_id), Some(LogState::Sealing { segment_index: current_segment_index, ..}) if segment_index == *current_segment_index)
+                {
+                    // todo debounce to avoid busy loop
+                    // todo what if it doesn't work again? Maybe stepping down as the leader
+                    effects.push(Effect::Seal {
+                        log_id,
+                        segment_index,
+                    })
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn on_logs_written(&mut self, version: Version) {
+        // filter out, outdated writes
+        if Some(version) == self.logs_write_in_progress {
+            self.logs_write_in_progress = None;
+        }
+    }
+
+    fn on_logs_update(&mut self, logs: Pinned<Logs>) -> Result<()> {
+        // rebuild the internal state if we receive a newer logs or one with a version we were
+        // supposed to write (race condition)
+        if logs.version() > self.current_logs.version()
+            || Some(logs.version()) == self.logs_write_in_progress
+        {
+            self.logs_write_in_progress = None;
+            self.current_logs = logs.into_arc();
+
+            // only keep provisioning logs since they are derived from the partition table
+            self.logs_state
+                .retain(|_, state| matches!(state, LogState::Provisioning { .. }));
+
+            for (log_id, chain) in self.current_logs.iter() {
+                let tail = chain.tail();
+
+                if let Some(seal_lsn) = tail.tail_lsn {
+                    // sealed tail segment
+                    self.logs_state.insert(
+                        *log_id,
+                        LogState::Sealed {
+                            configuration: tail.config.try_into().map_err(|err| {
+                                LogsControllerError::LogletParamsToConfiguration(err)
+                            })?,
+                            segment_index: tail.index(),
+                            seal_lsn,
+                        },
+                    );
+                } else {
+                    // open tail segment
+                    self.logs_state.insert(
+                        *log_id,
+                        LogState::Available {
+                            configuration: Some(tail.config.try_into().map_err(|err| {
+                                LogsControllerError::LogletParamsToConfiguration(err)
+                            })?),
+                            segment_index: tail.index(),
+                        },
+                    );
+                }
+                // todo how to figure out whether a previous logs controller instance has started
+                //  sealing a loglet? Can we ask the log servers about it? Should we periodically
+                //  check via bifrost?
```

Comment:
The first case should be handled currently, because it would eventually result in a new `Logs` being detected. So this would mean that we start for every log a periodic check which runs `find_tail`, I guess.